docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -667,3 +671,188 @@ Delegates votes from the sender to `delegatee`.
+
+NOTE: This value is the sum of all available votes, which is not necessarily the sum of all delegated votes.
+Votes that have not been delegated are still part of total supply, even though they would not participate in a
+vote.
+
+Requirements:
+
+- `timepoint` must be in the past.
+
+[.contract-item]
+[[VotesComponent-delegates]]
+==== `[.contract-item-name]#++delegates++#++(self: @ComponentState<TContractState>, account: ContractAddress) → ContractAddress++` [.item-kind]#external#
+
+Returns the delegate that `account` has chosen.
+
+[.contract-item]
+[[VotesComponent-delegate]]
+==== `[.contract-item-name]#++delegate++#++(ref self: ComponentState<TContractState>, delegatee: ContractAddress)++` [.item-kind]#external#
+
+Delegates votes from the sender to `delegatee`.
+
+Emits a {DelegateChanged} event.
+
+May emit one or two {DelegateVotesChanged} events.
+
+[.contract-item]
+[[VotesComponent-delegate_by_sig]]
+==== `[.contract-item-name]#++delegate_by_sig++#++(ref self: ComponentState<TContractState>, delegator: ContractAddress, delegatee: ContractAddress, nonce: felt252, expiry: u64, signature: Array<felt252>)++` [.item-kind]#external#
+
+Delegates votes from `delegator` to `delegatee` through a SNIP12 message signature validation.
+
+Requirements:
+
+- `expiry` must not be in the past.
+- `nonce` must match the account's current nonce.
+- `delegator` must implement `SRC6::is_valid_signature`.
+- `signature` should be valid for the message hash.
+
+Emits a {DelegateChanged} event.
+
+May emit one or two {DelegateVotesChanged} events.
+
+[#VotesComponent-Internal-functions]
+==== Internal functions
+
+[.contract-item]
+[[VotesComponent-_delegate]]
+==== `[.contract-item-name]#++_delegate++#++(ref self: ComponentState<TContractState>, account: ContractAddress, delegatee: ContractAddress)++` [.item-kind]#internal#
+
+Delegates all of ``account``'s voting units to `delegatee`.
+
+Emits a {DelegateChanged} event.
+
+May emit one or two {DelegateVotesChanged} events.
+
+[.contract-item]
+[[VotesComponent-move_delegate_votes]]
+==== `[.contract-item-name]#++move_delegate_votes++#++(ref self: ComponentState<TContractState>, from: ContractAddress, to: ContractAddress, amount: u256)++` [.item-kind]#internal#
+
+Moves delegated votes from one delegate to another.
+
+May emit one or two {DelegateVotesChanged} events.
+
+[.contract-item]
+[[VotesComponent-transfer_voting_units]]
+==== `[.contract-item-name]#++transfer_voting_units++#++(ref self: ComponentState<TContractState>, from: ContractAddress, to: ContractAddress, amount: u256)++` [.item-kind]#internal#
+
+Transfers, mints, or burns voting units.
+
+To register a mint, `from` should be zero. To register a burn, `to`
+should be zero. Total supply of voting units will be adjusted with mints and burns.
+
+Special care must be taken to ensure that this function is called correctly. In `ERC20` and `ERC721` this is done usually via the hooks.
```

Comment:
```suggestion
Special care must be taken to ensure that this function is called correctly. In `ERC20` and `ERC721` components this is done usually via the hooks.
```

Reply:
Well... for people implementing this won't be a component, it will be a contract in any case. What do you think about changing components for contracts?